lambda_src/drivers/destination_s3.py

```diff
@@ -107,9 +109,11 @@ def write(
     )
 
     if not prefix:
-        prefixed_filename = f'{DATA_FOLDER_NAME}/{batch_id}/row-{row_index}.data.json'
+        prefixed_filename = f'{batch_id}-row-{row_index}.data.json'
+    elif prefix.endswith('/'):
+        prefixed_filename = f'{prefix}{batch_id}-row-{row_index}.data.json'
```

Comment:
probably best to separate batch and row with `_` instead of `-row-`, that's how AWS does it

Reply:
Ah okay, will change it.